test/recipes/30-test_evp.t

```diff
@@ -47,8 +47,8 @@ my @files = qw(
                 evpmd_sha.txt
                 evppbe_pbkdf2.txt
                 evppkey_dsa.txt 
-                evppkey_ecc_nist.txt
-                evppkey_ecdh_nist.txt
+                evppkey_ecc.txt
+                evppkey_ecdh.txt
```

Comment:
When I did the split back in #12672, I failed to recognise that `test/evp_test.c` does have a mechanism to see if a pkey couldn't be loaded because of unsupported algorithms, see `key_unsupported()`, and that the problem I saw then was due to shadowed error codes.  So here, I'm reverting that change and correcting error reporting instead.

Otherwise, I fear that we'll go nuts having to remember which EC keys go in what file, and even worse when standards change going forward and all moving EC keys around that would follow.